Index: src/c/src/zookeeper.c
===================================================================
--- src/c/src/zookeeper.c       (revision 1173876)
+++ src/c/src/zookeeper.c       (working copy)
@@ -1543,7 +1543,9 @@
     gettimeofday(&now, 0);
     if(zh->next_deadline.tv_sec!=0 || zh->next_deadline.tv_usec!=0){
         int time_left = calculate_interval(&zh->next_deadline, &now);
-        if (time_left > 10)
+        int max_exceed = zh->recv_timeout / 10 > 200 ? 200 :
+                         (zh->recv_timeout / 10);
+        if (time_left > max_exceed)
             LOG_WARN(("Exceeded deadline by %dms", time_left));
     }
     api_prolog(zh);
